rfcs/websockets_http2.md

```diff
@@ -0,0 +1,24 @@
+# RFC #67: Add WebSockets over HTTP/2 support to wptserve
+
+## Summary
+
+Enable wptserve to support WebSockets over an existing http2 connection.
+
+## Details
+
+Allow tests to open a WebSocket over an existing http2 connection.
+
+wptserver needs to implement the [Bootstrapping WebSocket with HTTP/2 protocol](https://tools.ietf.org/html/rfc8441).
+
+It means that it needs to be modified to:
+- Set ENABLE_CONNECT_PROTOCOL settings to 1
+- Detect Extended CONNECT Method and implement the WebSocket opening handshake.
+- pywebsocket operates on Apache mod_python Request objects. So we need to wrap H2Response and H2Request into a Request-like object.
+
+## Risks
+
+WebSocket tests who source **websockets/constants.js** have a wss variant. We will need to add a new variant named **http2** in order to use the http2 port when opening the WebSockets. WTP server does not support insecure HTTP/2 so we don't need to expose an insecure HTTP/2 WebSocket variant.
+
+This change needs to update hyper libraries to a more recent version.
```

Comment:
Are there notable changes in the update that might either affect existing tests or require installing new third-party libraries (especially any that require non-Python code)

Reply:
nothing of that sort.

After double checking hyper changelog though, I notice that in the version we need, support for python 3.3 and python 3.7 is dropped.

Does wpt support those versions of python? If so, the alternative is to backport the 3 patches we need.

Reply:
I believe we are targeting the range 3.6-3.8 initially. I wonder why 3.7 support was dropped; it seems to still be an officially supported version of Python.

Reply:
Sorry my bad. I had misread hyper changelog. It actually says:
"Drop support for Python 3.3 and support 3.7"


Since wpt support 3.6-3.8, this is fine.